readalongs/cli.py

```diff
@@ -66,116 +73,132 @@ def cli():
+@click.option("-f", "--force-overwrite", is_flag=True, help="Force overwrite output files")
+@click.option("-i", "--text-input", is_flag=True, help="Input is plain text (assume paragraphs separated by blank lines, 1 paragraph per page)")
+@click.option("-l", "--language", type=click.Choice(LANGS, case_sensitive=False), help="Set language for plain text input")
+@click.option("-u", "--unit", type=click.Choice(["w", "m"], case_sensitive=False), help="Unit (w = word, m = morpheme) to align to")
+@click.option("-s", "--save-temps", is_flag=True, help="Save intermediate stages of processing and temporary files (dictionary, FSG, tokenization etc)")
+@click.option("-t", "--text-grid", is_flag=True, help="Export to Praat TextGrid & ELAN eaf file")
+@click.option("-x", "--output-xhtml", is_flag=True, help="Output simple XHTML instead of XML")
 def align(**kwargs):
-    """Align INPUTFILE and WAVFILE and create output files at OUTPUT_BASE.
+    """Align INPUTFILE and AUDIOFILE and create output files at OUTPUT_BASE.
 
     inputfile : A path to the input text file
 
-    wavfile : A path to the input audio file
+    audiofile : A path to the input audio file
 
     output-base : A base name for output files
     """
-    if os.path.exists(kwargs['output_base']):
-        if not kwargs['force_overwrite']:
+    config = kwargs.get("config", None)
+    if config and config.endswith("json"):
+        try:
+            with open(config) as f:
+                config = json.load(f)
+        except json.decoder.JSONDecodeError:
+            LOGGER.error(f"Config file at {config} is not valid json.")
+
+    if os.path.exists(kwargs["output_base"]):
+        if not kwargs["force_overwrite"]:
             raise click.UsageError(
-                f"Output folder '{kwargs['output_base']}' already exists")
+                f"Output folder '{kwargs['output_base']}' already exists"
+            )
     else:
-        os.mkdir(kwargs['output_base'])
-    output_base = os.path.join(kwargs['output_base'], os.path.basename(kwargs['output_base']))
-    if kwargs['debug']:
-        LOGGER.setLevel('DEBUG')
-    if kwargs['text_input']:
-        if not kwargs['language']:
-            LOGGER.warn(f"No input language provided, using undetermined mapping")
-        tempfile, kwargs['inputfile'] \
-            = create_input_tei(kwargs['inputfile'],
-                               text_language=kwargs['language'],
-                               save_temps=(output_base
-                                           if kwargs['save_temps'] else None))
-    if kwargs['output_xhtml']:
-        tokenized_xml_path = '%s.xhtml' % output_base
+        os.mkdir(kwargs["output_base"])
+    output_base = os.path.join(
+        kwargs["output_base"], os.path.basename(kwargs["output_base"])
+    )
+    if kwargs["debug"]:
+        LOGGER.setLevel("DEBUG")
+    if kwargs["text_input"]:
+        if not kwargs["language"]:
+            LOGGER.warn(
```

Comment:
I'm really not that keen on adding these extra newlines, like here, and above where you have `(` on one line, the `f"..."` string on another, and `)` on yet another. I'm not going to fight it, but I hope you don't mind if I don't systematically reproduce it in my own code...

Reply:
I think this goes back to deciding on an auto-formatting tool. This was done by `black` on my machine